src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/AuditLog.java

```diff
@@ -69,13 +69,15 @@
     void logSSLException(TransportRequest request, Throwable t, String action, Task task);
     void logSSLException(RestRequest request, Throwable t);
 
-    void logDocumentRead(String index, String id, ShardId shardId, Map<String, String> fieldNameValues, ComplianceConfig complianceConfig);
-    void logDocumentWritten(ShardId shardId, GetResult originalIndex, Index currentIndex, IndexResult result, ComplianceConfig complianceConfig);
+    void logDocumentRead(String index, String id, ShardId shardId, Map<String, String> fieldNameValues);
+    void logDocumentWritten(ShardId shardId, GetResult originalIndex, Index currentIndex, IndexResult result);
     void logDocumentDeleted(ShardId shardId, Delete delete, DeleteResult result);
     void logExternalConfig(Settings settings, Environment environment);
     
     // compliance config
     void setComplianceConfig(ComplianceConfig complianceConfig);
```

Comment:
Is it necessary to have `setComplianceConfig()` method? Can it be instantiated inside `AbstractAuditLog` constructor?

Reply:
I want to retain this method as subscribing for changes should be as simple as adding @Subscribe to the function for changes.

Reply:
`@Subscribe` should not be on the interface method. `NullAuditLog` should not subscribe to the event bus.

Reply:
Sorry I meant the function in AuditLogImpl and not this interface method